### PR TITLE
Added build specific test artifact deployment

### LINKED ISSUE #450: Add build specific artefact uplaod to build process
At the moment swtbot screen captures are uploaded to Sourceforge. A new build is overwriting these images.

What si required is a build specific zip which will additionally contain aretfacts such as the log outputs when tests are executed.

### PR DESCRIPTION
**Pull request by ghost**
- SWTBOT Artifacts are now zipped and uploaded to sourceforge
- zip files are individually named, thus they don't get overwritten
- Updated scripts so that maven script can be run locally on Linux computers
- Updated deploy script, so it only deploys in case of test artifacts exist
- Changed strategy on codecov reporting to try to send reports as early as possible. which means also on failed test cases. This may improve finding runtime issues on failing test cases due to race conditions.

Closes #450

### CODE REVIEW

File: bash/deploy_sourceforge.sh [hunk 1]
```diff
@@ -33,8 +33,22 @@ printUsage() {
...
-	rsync -e ssh -avP $TRAVIS_BUILD_DIR/swtbot/  dlrscmns@frs.sourceforge.net:/home/frs/project/virtualsatellite/VirtualSatellite4-Core/swtbot/
+	if [ "$(ls -A ${TRAVIS_BUILD_DIR}/swtbot)" ]; then
+		echo "Test Artifacts: Collecting all reports in: ${TRAVIS_BUILD_DIR}/swtbot"
+		cp $TRAVIS_BUILD_DIR/de.dlr.sc.virsat.swtbot.test/target/surefire-reports/* $TRAVIS_BUILD_DIR/swtbot/
+		TEST_ARTEFACTS_ZIP=SwtBot_${TRAVIS_REPO_SLUG}_${TRAVIS_BRANCH}_${TRAVIS_JOB_NUMBER}_${TRAVIS_COMMIT}
+		echo "Test Artifacts: About to zip all artifacts into: ${TEST_ARTEFACTS_ZIP}"
+		TEST_ARTEFACTS_ZIP=$(sed -e 's/\//\_/g' <<< ${TEST_ARTEFACTS_ZIP})
+		echo "Test Artifacts: About to zip all artifacts into: ${TEST_ARTEFACTS_ZIP}"
+		TEST_ARTEFACTS_ZIP=$(sed -e 's/\./\_/g' <<< ${TEST_ARTEFACTS_ZIP}).zip
+		echo "Test Artifacts: About to zip all artifacts into: ${TEST_ARTEFACTS_ZIP}"
```
Comment (andriikovalov-dlr) [inline bash/deploy_sourceforge.sh]: Let's remove intermediate debug echoes

File: bash/deploy_sourceforge.sh [hunk 2]
```diff
@@ -33,8 +33,22 @@ printUsage() {
...
-	cp $TRAVIS_BUILD_DIR/de.dlr.sc.virsat.swtbot.test/target/surefire-reports/* $TRAVIS_BUILD_DIR/swtbot/
-	rsync -e ssh -avP $TRAVIS_BUILD_DIR/swtbot/  dlrscmns@frs.sourceforge.net:/home/frs/project/virtualsatellite/VirtualSatellite4-Core/swtbot/
+	if [ "$(ls -A ${TRAVIS_BUILD_DIR}/swtbot)" ]; then
+		echo "Test Artifacts: Collecting all reports in: ${TRAVIS_BUILD_DIR}/swtbot"
+		cp $TRAVIS_BUILD_DIR/de.dlr.sc.virsat.swtbot.test/target/surefire-reports/* $TRAVIS_BUILD_DIR/swtbot/
+		TEST_ARTEFACTS_ZIP=SwtBot_${TRAVIS_REPO_SLUG}_${TRAVIS_BRANCH}_${TRAVIS_JOB_NUMBER}_${TRAVIS_COMMIT}
+		echo "Test Artifacts: About to zip all artifacts into: ${TEST_ARTEFACTS_ZIP}"
+		TEST_ARTEFACTS_ZIP=$(sed -e 's/\//\_/g' <<< ${TEST_ARTEFACTS_ZIP})
+		echo "Test Artifacts: About to zip all artifacts into: ${TEST_ARTEFACTS_ZIP}"
+		TEST_ARTEFACTS_ZIP=$(sed -e 's/\./\_/g' <<< ${TEST_ARTEFACTS_ZIP}).zip
```
Comment (andriikovalov-dlr) [inline bash/deploy_sourceforge.sh]: Can we maybe substitute both / and . in one sed command?

**Review by andriikovalov-dlr — CHANGES_REQUESTED** [review]
I'm wondering if we are deploying all swtbot results then who is going to delete them?

File: bash/deploy_sourceforge.sh (hunk 1 above)
Comment (PhilMFischer) [reply bash/deploy_sourceforge.sh]: Yes tried it. Have a look

File: bash/deploy_sourceforge.sh (hunk 2 above)
Comment (PhilMFischer) [reply bash/deploy_sourceforge.sh]: Tried it as well. seems to work on a local Linux machine.

**Review by SaMuellerDLR — APPROVED** [review]
Approved!

**Review by andriikovalov-dlr — APPROVED** [review]

**Pull request merged**